jobs/haproxy/templates/haproxy_wrapper.erb

```diff
@@ -84,13 +81,28 @@ trap shutdown QUIT INT TERM EXIT
 # enable hitless reloads
 trap reload_daemon USR2
 
+<%-
+  flags=[]
+
+  # Enable background mode only if syslog is not configured to output to stdout/stderr
+  run_in_foreground = p('ha_proxy.syslog_server') == "stdout" || p('ha_proxy.syslog_server') == "stderr"
+
+  if run_in_foreground
+    flags += ['-V', '-db']
+  else
+    flags += ['-D']
+  end
+
+  if p('ha_proxy.master_cli_enable')
+    flags += ['-S', p('ha_proxy.master_cli_bind')]
+  end
+-%>
+
 # Start up
 echo "$(date): Starting HAProxy"
-<%- if p('ha_proxy.syslog_server') == "stdout" || p('ha_proxy.syslog_server') == "stderr" -%>
-"${DAEMON}" -f "${CONFIG}" -W -p ${PID_FILE} -V -db
-<%- else -%>
-"${DAEMON}" -f "${CONFIG}" -W -D -p ${PID_FILE}
+haproxy -f /var/vcap/jobs/haproxy/config/haproxy.config -W -p ${PID_FILE} <%= flags.join(' ') %>
```

Comment:
why keep PID_FILE in a variable but hard-code CONFIG?
I'd rather keep the CONFIG variable, too

Reply:
NP I've changed it back. My original reasoning was that a variable that is only used once is an unnecessary abstraction. PID_FILE was used in a few different places so worth extracting into a variable.